cmd/kubernetes-static/main.go

```diff
@@ -55,58 +54,47 @@ func main() {
 	}
 
 	fakeK8s := fake.NewSimpleClientset(testutil.K8sEverything()...)
-	integration, err := integration.New(integrationName, integrationVersion, integration.Args(&args))
+	i, err := integration.New(integrationName, integrationVersion, integration.Args(&args))
 	if err != nil {
 		logrus.Fatal(err)
 	}
 
 	logger := log.NewStdErr(args.Verbose)
 
-	// ApiServer
-	apiServerClient := apiserver.TestAPIServer{Mem: map[string]*apiserver.NodeInfo{
-		// this nodename should be the same as the ones in the data folder
-		"minikube": {
-			NodeName: "minikube",
-			Labels: map[string]string{
-				"node-role.kubernetes.io/master": "",
-			},
-			Conditions: []v1.NodeCondition{
-				{
-					Type:   "DiskPressure",
-					Status: v1.ConditionFalse,
-				},
-				{
-					Type:   "MemoryPressure",
-					Status: v1.ConditionFalse,
-				},
-				{
-					Type:   "DiskPressure",
-					Status: v1.ConditionFalse,
-				},
-				{
-					Type:   "PIDPressure",
-					Status: v1.ConditionFalse,
-				},
-				{
-					Type:   "Ready",
-					Status: v1.ConditionTrue,
-				},
-			},
-			Unschedulable: false,
-		},
-	}}
+	nodeGetter, closeChan := discovery.NewNodeLister(fakeK8s)
+	defer close(closeChan)
+
+	u, err := url.Parse(testSever.KubeletEndpoint())
+	if err != nil {
+		log.Fatal(err)
+	}
 
 	// Kubelet
-	kubeletClient := newBasicHTTPClient(testSever.KubeletEndpoint())
+	mc := kubletClient.MockConnector{
+		URL:    *u,
+		Client: &http.Client{Timeout: time.Minute * 10},
+		Err:    nil,
+	}
+	kubeletClient, err := kubletClient.New(nil, config.Mock{}, &rest.Config{}, kubletClient.WithCustomConnector(mc))
+	if err != nil {
+		log.Fatal(err)
+	}
+
```

Comment:
wouldn't be possible to create the scraper for kubelet after this point? i'm a bit concerned that we have some logic in the scraper that needs to be reimplemented here, and also to make this main as simple and similar to main as possible.

Reply:
Actually it should never fail, since it is using the mock connector locally. However, if it fails there should be for sure a mistake in the way the mock client was populated

Reply:
regarding refactoring the static-main I agree, but I would do it in a different pr :)